services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedTenantService.java

```diff
@@ -360,64 +357,56 @@ private void addTenant(final JsonObject tenantToAdd) {
         Objects.requireNonNull(tenantId);
         Objects.requireNonNull(resourceVersion);
 
-        return Future.succeededFuture(removeTenant(tenantId, resourceVersion, span));
-    }
-
-    Result<Void> removeTenant(final String tenantId, final Optional<String> resourceVersion, final Span span) {
-
-        Objects.requireNonNull(tenantId);
-
         if (getConfig().isModificationEnabled()) {
             if (tenants.containsKey(tenantId)) {
                 final String actualVersion = tenants.get(tenantId).getVersion();
                 if (checkResourceVersion(resourceVersion, actualVersion)) {
                     tenants.remove(tenantId);
                     dirty.set(true);
-                    return Result.from(HttpURLConnection.HTTP_NO_CONTENT);
+                    return Future.succeededFuture(Result.from(HttpURLConnection.HTTP_NO_CONTENT));
                 } else {
                     TracingHelper.logError(span, "Resource Version mismatch.");
-                    return Result.from(HttpURLConnection.HTTP_PRECON_FAILED);
+                    return Future.failedFuture(new ClientErrorException(
+                            tenantId,
+                            HttpURLConnection.HTTP_PRECON_FAILED,
+                            "resource Version mismatch"));
                 }
             } else {
                 TracingHelper.logError(span, "Tenant not found.");
-                return Result.from(HttpURLConnection.HTTP_NOT_FOUND);
+                return Future.failedFuture(new ClientErrorException(
+                        tenantId,
+                        HttpURLConnection.HTTP_NOT_FOUND));
```

Comment:
How about also adding a corresponding error message in case of _HTTP_NOT_FOUND_ ? So that the error json has some meaningful message instead of `N/A`.

Reply:
I have pushed corresponding changes